core/modules/layout/layout.flexible.inc

```diff
@@ -488,24 +488,71 @@ function layout_flexible_template_edit_row_form($form, &$form_state, LayoutFlexi
 
   $form['region_names'] = array(
     '#type' => 'fieldset',
-    '#title' => t('Region names'),
+    '#title' => t('Regions'),
     '#collapsed' => $original_row != 'add',
     '#collapsible' => TRUE,
     '#tree' => TRUE,
   );
 
   $form_state['last_region_number'] = layout_flexible_template_get_last_region_number($flexible_template);
   for ($i = 0; $i < $region_count; $i++) {
-    $form['region_names']['region_name_' . $i] = array(
+    $form['region_names']['region_' . $i] = array(
+      '#type' => 'container',
+    );
+    if (isset($row_data['region_names']['region_' . $i]['region_name'])) {
+      $region_name = $row_data['region_names']['region_' . $i]['region_name'];
+    }
+    $form['region_names']['region_' . $i]['region_name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Region name') . ' ' . ($i + 1),
+      '#default_value' => !empty($region_name) ? $region_name : t('Region') . ' ' . ($form_state['last_region_number'] + $i + 1),
+    );
+    $form['region_names']['region_' . $i]['class_name'] = array(
+      '#type' => 'machine_name',
+      '#description' => t('A machine name to be used as the default CSS class. It must only contain uppercase letters, lowercase letters, numbers, hyphens and underscores.'),
+      '#default_value' => !empty($row_data['region_names']['region_' . $i]['class_name']) ? $row_data['region_names']['region_' . $i]['class_name'] : '',
+      '#maxlength' => 255,
+      '#field_prefix' => '<span dir="ltr">l-region--',
+      '#field_suffix' => '</span>&lrm;',
+      '#size' => 15,
+      '#disabled' => FALSE,
+      '#machine_name' => array(
+        'exists' => '_layout_flexible_template_region_class_name_exists',
+        'source' => array('region_names', 'region_' . $i, 'region_name'),
+        'standalone' => TRUE,
+        'label' => t('CSS class'),
```

Comment:
I think expanding the purpose of the machine name beyond just "CSS class" would be valuable for future changes. For example we may find it desirable to use the machine name in config files to reference where blocks are positioned, rather than a UUID. So this may be better labeled as "Machine name", "Internal name", or simply "Name" (while keeping the same description).

```suggestion
        'label' => t('Name'),
```